File: web/src/api/cloud/volume.js
import service from '@/utils/request'

// GetVolumes 获取卷列表
export const getVolumes = (params) => {
  return service({
    url: '/cloud/volume/list',
    method: 'get',
    params
  })
}

// CreateVolume 创建卷
export const createVolume = (data) => {
  return service({
    url: '/cloud/volume/create',
    method: 'post',
    data
  })
}

// RemoveVolume 删除卷
export const removeVolume = (data) => {
  return service({
    url: '/cloud/volume/delete',
    method: 'delete',
    data
  })
}

File: server/router/cloud/volume.go
package cloud

import (
	"github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type VolumeRouter struct{}

func (r *VolumeRouter) InitVolumeRouter(Router *gin.RouterGroup) {
	volumeRouter := Router.Group("cloud/volume").Use(middleware.OperationRecord())
	volumeRouterWithoutRecord := Router.Group("cloud/volume")
	var volumeApi = v1.ApiGroupApp.CloudApiGroup.VolumeApi
	{
		volumeRouter.POST("create", volumeApi.CreateVolume)  // 创建卷
		volumeRouter.DELETE("delete", volumeApi.RemoveVolume) // 删除卷
	}
	{
		volumeRouterWithoutRecord.GET("list", volumeApi.GetVolumes) // 获取卷列表
	}
}
